hs-bindgen/src/HsBindgen/Hs/Translation.hs

```diff
@@ -568,6 +571,47 @@ macroExprHsExpr = goExpr where
     goInt (C.IntegerLiteral { integerLiteralType = mbIntTy, integerLiteralValue = i }) =
       Just $ Hs.VarDeclIntegral i (maybe HsPrimCInt (uncurry integralType) mbIntTy)
 
+    goChar :: C.CharLiteral -> Maybe (Hs.VarDeclRHS ctx)
+    goChar (C.CharLiteral { charLiteralValue = c }) =
+      ( `Hs.VarDeclIntegral` HsPrimCInt ) <$>
+        case c of
+          C.CodeUnit u
```

Comment:
extract into separate functions. Preferably don't invent UTF8 parsing validation, there are plenty in the libs.